spec/javascript/utils/compiler/variables.test.js

```diff
@@ -397,5 +394,68 @@ describe("variables.js", () => {
           .map((name) => [name, getDefaultVariableNameIndex(name)]))
       ).toStrictEqual(expectedOutput)
     })
+
+    test("Should get Global variables from parameter objects", () => {
+      completionsMap.set(setCompletionsMap("For Global Variable"))
+      const input = "For Global Variable({ One: variable1, Two: Global.variable2, Three: Count Of(Array()), Four: 1 })"
+      const expectedOutput = {
+        globalVariables: ["variable2", "variable1"],
+        playerVariables: []
+      }
+  
+      expect(getVariables(input)).toEqual(expectedOutput)
+    })
+  
+    test("Should get variables from nested parameter objects", () => {
+      completionsMap.set(setCompletionsMap("For Global Variable"))
+      const input =
+        `For Global Variable({
+          One: variable1,
+          Two: 0,
+          Three: Some Action({ 
+            First: Global.variable2,
+            Second: Event Player.variable3
+          }),
+          Four: 1
+        })`
+      const expectedOutput = {
+        globalVariables: ["variable2", "variable1"],
+        playerVariables: ["variable3"]
+      }
+  
+      expect(getVariables(input)).toEqual(expectedOutput)
+    })
+  
+    test("Should get Player variables from parameter objects", () => {
+      completionsMap.set(setCompletionsMap("Chase Player Variable Over Time"))
+      const input =
+        `Chase Player Variable Over Time({ 
+          One: Event Player,
+          Two: variable1,
+          Three: 0,
+          Four: 1,
+          Five: Destination And Duration
+        })`
+      const expectedOutput = {
+        globalVariables: [],
+        playerVariables: ["variable1"]
+      }
+  
+      expect(getVariables(input)).toEqual(expectedOutput)
+    })
   })
 })
+
+function setCompletionsMap(actionLabel) {
```

Comment:
This isn't really `set`ting `CompletionsMap`, but rather generating its content. How about a name like "createCompletionMapItems"?

Though I'd guess this function becomes irrelevant if you are moving everything to a `beforeAll()` block.

Reply:
Better yet, move it to a constant at the start of the file. Writing function in your tests gets "dangerous" as you will need to start writing tests for your test. Write the value to a single constant and when you need to change the label do `constant.label = "new value"`

Reply:
isnt that more or less equivalent to what i am already doing, except im setting it in a function? the only thing im changing is the label, the rest is constant.
but yeah the name is not good, i was actually going to change it but i forgor 💀